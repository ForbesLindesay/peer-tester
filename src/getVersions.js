import {inspect} from 'util';
import {satisfies} from 'semver';
import RegClient from 'npm-registry-client';

const client = new RegClient({});

const params = {timeout: 1000};

function getVersions(name, range) {
  return new Promise((resolve, reject) => {
    if (typeof name !== 'string') {
      throw new Error('Expected name to be a string but got ' + inspect(name));
    }
    client.get('https://registry.npmjs.org/' + name, params, function (err, data, raw, res) {
      if (err) return reject(err);
      const versions = Object.keys(data.versions || {});
      const satisfyingVersions = versions.filter(version => satisfies(version, range));
      resolve(satisfyingVersions);
    });
  });
}

export default getVersions;
