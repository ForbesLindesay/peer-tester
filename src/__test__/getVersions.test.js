import getVersions from '../getVersions';

test('getVersions("react", "^0.14.0")', async () => {
  const versions = await getVersions('react', '^0.14.0');
  expect(versions).toMatchSnapshot();
});
