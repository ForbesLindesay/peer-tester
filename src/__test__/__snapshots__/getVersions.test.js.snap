// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getVersions("react", "^0.14.0") 1`] = `
Array [
  "0.14.0",
  "0.14.1",
  "0.14.2",
  "0.14.3",
  "0.14.4",
  "0.14.5",
  "0.14.6",
  "0.14.7",
  "0.14.8",
  "0.14.9",
]
`;
